src/lib/input-validation/utils/regexGenerator.ts

```diff
@@ -235,8 +235,9 @@ export const generatePattern = ({
   let allowedChars = allowedCharacterString;
   if (!!allowedSpecialChars) {
     if (allowedCharacterString !== "") {
+      const quantifier = specialCharacterLocation === CharacterLocationEnum.anywhere ? "*" : "+";
```

Comment:
using + is slow. we need make some adjustment in consumer app when `specialCharacterLocation` is `middle`